💎(site) Case Studies Module

nickalcantara opened the pull request:
Introduces a Case Studies module for the homepage, consisting of a layout component and a Study Card component. I took an intrinsic approach, but the Study Card component's design changes in several ways that weren't compatible with [R.A.M](https://web.dev/patterns/layout/repeat-auto-minmax/) or other related patterns, prompting the use of breakpoints that were identified via manual testing.

- Resolves #211

nickalcantara committed:
case studies, study card layout

nickalcantara committed:
study card responsiveness

nickalcantara committed:
resolve conflicts, linting

nickalcantara committed:
intrinsic approach refactor

nickalcantara committed:
Merge branch 'redesign/2022' of https://github.com/chromeos/chromeos.dev into feature/case-studies

nickalcantara committed:
VQA feedback, linting

nickalcantara committed:
use l10n.locale for cta, add docs

nickalcantara committed:
update docs

Snugug commented on site/_components/case-studies.njk:
```diff
@@ -0,0 +1,31 @@
+    {% endif %}
+  </header>
+  <div class="case-studies__content">{% for study in studies %}{{cardStudy(study, loop.first)}}{% endfor %}</div>
+  {% if content.cta %} {% set localeCta = { 'url': '/' ~ locale ~ '/' ~ content.cta.url, 'text': content.cta.text }%}
```
I have other comments about how to handle this in other PRs; let's align on one (maybe as a separate PR) and update all accordingly

Snugug commented on site/_data/eleventyComputed.js:
```diff
@@ -305,13 +305,25 @@ module.exports = {
         .sort(propSort({ prop: 'data.title', lowercase: true }))
         .sort(propSort({ prop: 'data.weight', fallback: 0 }))
         .map((i) => ({
+          eyebrow: get(i, 'data.tags[0]'),
```
```suggestion
          eyebrow: get(i, 'data.tags[1]'),
```

Because we use [data directories](https://www.11ty.dev/docs/data-template-dir/) as part of our 11ty data cascade to insert `case study` into each of our case studies, the we should use the second item, which is going to be the first descriptive tag of the case study itself

Snugug commented on site/_data/eleventyComputed.js:
```diff
@@ -305,13 +305,25 @@ module.exports = {
+            name: get(i, 'data.app.name'),
+            company: get(i, 'data.app.company'),
+          },
+          media: {
+            src: get(i, 'data.hero.image'),
+            alt: get(i, 'data.hero.alt'),
+          },
```
Media should pull from `data.featured` because items that get featured may have different featuring images than hero images that are used in the post itself. I believe we already have computed data to uplevel `hero` into `featured` if there isn't a featured image

Snugug commented on site/_data/eleventyComputed.js:
```diff
@@ -305,13 +305,25 @@ module.exports = {
+          eyebrow: get(i, 'data.tags[0]'),
           title: get(i, 'data.title'),
           image: get(i, 'data.app.logo'),
+          logo: {
```
I think line 310 isn't needed anymore if you're grabbing the logo here

Snugug commented on site/sass/components/_card-study.scss:
```diff
@@ -0,0 +1,162 @@
+    margin-top: auto;
+  }
+
+  &--featured {
```
Given that featured case studies have such different styling from regular case studies, I recommend you make a separate block, `featured-case-study` (and maybe even a separate component) to differentiate between the two and make their styling journeys easier to follow.

Snugug commented on site/sass/components/_card-study.scss:
```diff
@@ -0,0 +1,162 @@
+      text-align: center;
+    }
+
+    @include mq($break-2-col) {
+      padding-bottom: 2rem;
+    }
```
```suggestion
```
Looking at the overall styling for this, your bottom padding is already `2rem` here thanks to line 35

Snugug commented on site/sass/components/_card-study.scss:
```diff
@@ -0,0 +1,162 @@
+    flex-direction: column;
+    padding: 2rem 1.5rem;
+
+    @include mq(0 419px) {
+      align-items: center;
+      text-align: center;
+    }
```
```suggestion
    text-align: center;
    align-items: center
    @include mq(419px) {
      align-items: revert;
      text-align: revert;
    }
```

Instead of clamping from 0 to a max-width like this, or using a `max-width` breakpoint, invert the styling so you only need a `min-width`. This ensures that our single column layout works without breakpoints, which should be our "mobile first" development goal as much as possible

Snugug commented on site/sass/components/_card-study.scss:
```diff
@@ -0,0 +1,162 @@
+  }
+
+  &--featured & {
+    $spacing-s: 32px;
```
```suggestion
    $spacing-s: 2rem;
```

Prefer `rem` directly instead of `px`

Snugug reviewed (changes requested):
Thanks for this work, and thanks for going the extra step and adding style guide entries for these elements! There are a few comments here cleaning up some code, but the two big updates to this that I'm looking for are:
* Splitting out Featured Case Study styling from the Study card because they're really two separate components even if they may share the same structure
* Looking at the intrinsic styling for the regular case study card again. I think you're right that they won't work well with Grid intrinsically, but their changes appear to work really well with Flexbox. I drew up a quick [Codepen](https://codepen.io/Snugug/pen/zYpxmej) to show the action in-place. Key items:
  * The image (here the "left" container) should grow but not shrink
  * The text (here the "right" container) should both grow and shrink
  * Their `flex-basis` should add up to 100%
  * Adding the `min-width` to the image is the key to having the items swap from vertical to horizontal
  * You probably need to remove the `height: 100%` from the image, too

As always, thanks for your work on this!

nickalcantara committed:
separate featured case study from default, convert to flex wrap pattern, update docs

nickalcantara committed:
Merge branch 'redesign/2022' of https://github.com/chromeos/chromeos.dev into feature/case-studies

nickalcantara commented:
@Snugug Thanks for your feedback - I've refactored to use the flex wrapping approach for the regular cards and will continue to use that pattern in similar components going forward! All other feedback was addressed with exception to the `$spacing-s` variable, which is in `px` units since its only used within our `generateScaling()` function and not directly. I've added a comment about this for clarity.

Vanss472 commented:
Hey @nickalcantara the conflicts you have here are based on my changes to the `eleventyComputed.js` that I mention to you yesterday and got merged last night. 

I'm fine with your changes, you might have to change the variables of this file `site/_components/featured-story.njk` Line 23 to reflect your changes. So instead of `image.image` it should be `media.image` and `media.alt`.

Incoming changes (should be ignore):
<img width="296" alt="image" src="https://user-images.githubusercontent.com/1170192/159725151-0950b0d9-209c-4e36-acc7-9171e8f525c6.png">

cc: @Snugug

nickalcantara committed:
Merge branch 'redesign/2022' of https://github.com/chromeos/chromeos.dev into feature/case-studies, update stories card data
